lib/Sema/TypeCheckType.cpp

```diff
@@ -5154,7 +5156,6 @@ NeverNullType TypeResolver::resolveImplicitlyUnwrappedOptionalType(
   case TypeResolverContext::TypeAliasDecl:
   case TypeResolverContext::GenericTypeAliasDecl:
   case TypeResolverContext::GenericRequirement:
-  case TypeResolverContext::ExistentialConstraint:
```

Comment:
How would you approach diagnosing `let _: (Int, any P!)`? In this case, the `!` is *actually* illegal, even if we write `(any P)!` (you cannot write `!` inside another type).